aws/resource_aws_lambda_invocation.go

```diff
@@ -0,0 +1,100 @@
+package aws
+
+import (
+	"crypto/md5"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceAwsLambdaInvocation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLambdaInvocationCreate,
+		Read:   resourceAwsLambdaInvocationRead,
+		Update: resourceAwsLambdaInvocationUpdate,
+		Delete: resourceAwsLambdaInvocationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"function_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"qualifier": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "$LATEST",
+			},
+
+			"input": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsJSON,
+			},
+
+			"invoke_on_update": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
```

Comment:
Using `triggers` with a map might be more consistent with other terraform (AWS) resources (and more powerful):
https://github.com/hashicorp/terraform-provider-aws/blob/5283e38ff9fa019c0d8aa3059a41a03cf431a2c2/aws/resource_aws_api_gateway_deployment.go#L45-L50

If I got it right, there would be no further checking (like in #L89) necessary anymore.